routes/userRoutes.js

```diff
@@ -0,0 +1,101 @@
+const express = require('express');
+const passport = require('passport');
+const validator = require('validator');
+
+const router = express.Router();
+const User = require('../models/User');
+
+/**
+ * @route   PUT user/:id
+ * @desc    update  User (name+avatar)
+ * @access  Private
+ */
+router.put('/:id', passport.authenticate('jwt', { session: false }), (req, res) =>
+	User.findById(req.params.id)
+		.then(user => {
+			if (!user) {
+				return res.status(404).json({ message: 'User not found' });
+			}
+			if (user._id.toHexString() !== req.user._id.toHexString()) {
+				return res.status(401).json({ message: 'Unauthorized' });
+			}
+
+			if (req.body.name) {
+				// Run validation for name
+				user.name = req.body.name;
+			}
+			// image URL validation
+			if (req.body.imgUrl) {
+				const validUrl = validator.isURL(req.body.imgUrl, {
+					protocols: ['http', 'https'],
+					require_protocol: true,
+					require_tld: true,
+				});
+				if (validUrl) {
+					user.avatar = req.body.imgUrl;
+				} else {
+					return res.status(400).json({ message: 'Bad image URL' });
+				}
+			};
+			if (req.body.about) {
+				user.about = req.body.about;
+			}
```

Comment:
Quicknote: We should handle a situation where no update values are provided and just return the user object straight away, or some error/warning. This is not critical as we're going to have some sort of validation on frontend so update req with no values to update shouldn't happen anyway.